### ISSUE TITLE
Export all accounts into multiple CSV files in one click

### ISSUE THREAD

**Issue by rmkhalfay** [moneymanagerex/moneymanagerex#2767]
# Description of the feature
Export all accounts into multiple CSV files in one click; if I have 20 accounts then it is cumbersome to do it one by one; Also JSON file import to ACCESS or EXCEL not very easy.  Also in JSON file exported did not see the field "BALANCE" and account balance is not exported to JSON file

Using latest version MMEX 1.3.6 64 bit
Windows 10

**Comment by vomikan** [comment]
I guess GRM may be used. All accounts may be saved into one file.

**Comment by rmkhalfay** [comment]
Not sure how GRM can be used? Can you help please

**Comment by TomAff** [comment]
I use sqlite3.exe, a batch file and a sql-script file

.bat
sqlite3.exe -header -csv -separator ";" privat.mmb < "sql_alldata.sql" > "export_alldata.csv"
sqlite3.exe -header -csv -separator ";" privat.mmb < "sql_categories.sql" > "export_categories.csv"
sqlite3.exe -header -csv -separator ";" privat.mmb < "sql_payees.sql" > "export_payees.csv"

it takes about 2 seconds and all I need is in separate csv-files

**Comment by rmkhalfay** [comment]
Sir can you help. How to use above script in GRM. Thanks

**Comment by TomAff** [comment]
Me? I do not use the GRM.

**Comment by rmkhalfay** [comment]
> I use sqlite3.exe, a batch file and a sql-script file
> 
> .bat
> sqlite3.exe -header -csv -separator ";" privat.mmb < "sql_alldata.sql" > "export_alldata.csv"
> sqlite3.exe -header -csv -separator ";" privat.mmb < "sql_categories.sql" > "export_categories.csv"
> sqlite3.exe -header -csv -separator ";" privat.mmb < "sql_payees.sql" > "export_payees.csv"
> 
> it takes about 2 seconds and all I need is in separate csv-files

Sir, Can you elaborate more on this; This batch file is to be created in MS Dos? or which software is required to run the above script; I am not familiar; I have used MS ACCESS with SQL queries in it; But how it can be used in MMEX data?

**Comment by vomikan** [comment]
GRM id General Report Manger.

https://github.com/moneymanagerex/general-reports

This report looks like suitable solution
https://github.com/moneymanagerex/general-reports/tree/master/packages/Transaction/Withdrawals

**Comment by vomikan** [comment]
Alternate way for export to Excel

![image](https://user-images.githubusercontent.com/6836805/125205985-74a26d00-e28d-11eb-8955-1c198acac41d.png)

save the results as test.html

Then run Excel and open test.html

![image](https://user-images.githubusercontent.com/6836805/125206033-aadfec80-e28d-11eb-8ea8-8e6594660774.png)

**Comment by rmkhalfay** [comment]
Many thanks for the prompt response. It is indeed very useful - if only I could transfer this way all the transactions (not only withdrawals) and paste in EXCEL. Yes you are correct in mentioning only some data can be exported to EXCEL

**Comment by TomAff** [comment]
I try to explain it step by step.

1. Donwload Sqlite3.exe from https://www.sqlite.org/index.html
2. copy sqlite3.exe in the directory where your MMEX-Database-File is saved.
3. Create a .bat-File. It's a simple text-file with ending .bat instead of .txt. It's easier than typing always a long script in the command line :)

sqlite3.exe -header -csv -separator ";" privat.mmb < "sql_alldata.sql" > "export_alldata.csv"

-header means fieldnames are exported in the first line
privat,mmb => your MMEX database file
< "sql_alldata.sql" => read sql-script from another file
> "export_alldata.csv" => write results into ...csv

you can try this in the command line in you db-directory
sqlite3.exe -header -csv -separator ";" YOURMMEXDATABATE.mmb "select * from category_v1" > "export_categories.csv"
this will export all your categories in a csv-file => just as an example

**Comment by TomAff** [comment]
I added a the "old" all-data-view in my database

select
       CANS.TransID as ID, CANS.TransCode as TransactionType,
       date(CANS.TransDate, 'localtime') as Date,
       d.userdate as UserDate
       ,coalesce(CAT.CategName, SCAT.CategName) as Category,
       coalesce(SUBCAT.SUBCategName, SSCAT.SUBCategName, '') as Subcategory,
       ROUND((case CANS.TRANSCODE when 'Withdrawal' then -1 else 1 end)*(case CANS.CATEGID when -1 then st.splittransamount else CANS.TRANSAMOUNT end),2) as Amount,
       cf.currency_symbol as currency,
       CANS.Status as Status, CANS.NOTES as Notes,
       FROMACC.CurrencyID as CurrencyID,
       FROMACC.AccountName as AccountName,  FROMACC.AccountID as AccountID,
       ifnull (TOACC.AccountName,'') as ToAccountName,
       ifnull (TOACC.ACCOUNTId,-1) as ToAccountID, CANS.ToTransAmount ToTransAmount,
       ifnull (TOACC.CURRENCYID, -1) as ToCurrencyID,
       (case ifnull(CANS.CATEGID,-1) when -1 then 1 else 0 end) as Splitted,
       ifnull(CAT.CategId,st.CategId) as CategID,
       ifnull (ifnull(SUBCAT.SubCategID,st.subCategId),-1) as SubCategID,
       ifnull (PAYEE.PayeeName,'') as Payee,
       ifnull (PAYEE.PayeeID,-1) as PayeeID,
            CANS.TRANSACTIONNUMBER as TransactionNumber,
            d.year as Year,
            d.month as Month,
            d.day as Day,
            d.finyear as FinYear,
            cfd.content as Document,
            att.description as attDocument,
            att.filename as attFilename
            from  CHECKINGACCOUNT_V1 CANS
            left join CATEGORY_V1 CAT on CAT.CATEGID = CANS.CATEGID
            left join SUBCATEGORY_V1 SUBCAT on SUBCAT.SUBCATEGID = CANS.SUBCATEGID and SUBCAT.CATEGID = CANS.CATEGID
            left join PAYEE_V1 PAYEE on PAYEE.PAYEEID = CANS.PAYEEID
            left join ACCOUNTLIST_V1 FROMACC on FROMACC.ACCOUNTID = CANS.ACCOUNTID
            left join ACCOUNTLIST_V1 TOACC on TOACC.ACCOUNTID = CANS.TOACCOUNTID
            left join splittransactions_V1 st on CANS.transid=st.transid
            left join CATEGORY_V1 SCAT on SCAT.CATEGID = st.CATEGID and CANS.TransId=st.transid
            left join SUBCATEGORY_V1 SSCAT on SSCAT.SUBCATEGID = st.SUBCATEGID and SSCAT.CATEGID = st.CATEGID and CANS.TransId=st.transid
            left join currencyformats_V1 cf on cf.currencyid=FROMACC .currencyid
            left join customfielddata_V1 cfd on cfd.REFID=cans.transid
            left join attachment_V1 att on att.REFID=cans.transid
            left join (select transid as id , date(transdate, 'localtime') as transdate
,round (strftime('%d', transdate, 'localtime'))  as day
,round (strftime('%m', transdate, 'localtime'))  as month
,round (strftime('%Y', transdate, 'localtime'))  as year
,round(strftime('%Y', transdate, 'localtime' ,'start of month', ((case when fd.infovalue<=round(strftime('%d', transdate , 'localtime')) then 1  else 0 end)-fm.infovalue)||' month')) as finyear
,ifnull (ifnull (strftime(df.infovalue, TransDate, 'localtime'),
       (strftime(replace (df.infovalue, '%y', SubStr (strftime('%Y',TransDate, 'localtime'),3,2)),TransDate, 'localtime'))
       ), date(TransDate, 'localtime')) as UserDate
from  CHECKINGACCOUNT_V1
left join infotable_V1 df on df.infoname='DATEFORMAT'
left join infotable_V1 fm on fm.infoname='FINANCIAL_YEAR_START_MONTH'
left join infotable_V1 fd on fd.infoname='FINANCIAL_YEAR_START_DAY') d on d.id=CANS.TRANSID
order by date desc, CANS.transid

**Comment by rmkhalfay** [comment]
Sir, 
Many thanks for the great help; It works like MAGIC; Sorry for troubling by asking one more question. I did not get the BALANCE FIELD in the exported CSV file; Is it possible? 
If not then no problems also - then I will need to put some formulae for each row to find out based on initial balance for each account and transaction type - Withdrawal/ Deposit / Transfer; 

Thank you again for this great help;

**Comment by TomAff** [comment]
Sorry, I do not now if there is a functions in sqlite. I think it is calulated in MMEX without SQL, so there is no field "balance" in the database.
In the alldata-view the field amount is shown as -/+ value.
Transfer could be a little tricky I think. I don't use transfer. I book it on a category like "money in transfer"

**Comment by tactilis** [comment]
>  I don't use transfer. I book it on a category like "money in transfer"

Out of interest, why do you choose to use this approach rather than using MMEX's Transfer transaction type?

What are the disadvantages for you of using Transfer (or the advantages of your method)?

**Comment by TomAff** [comment]
Depending on the bank, the payment may be made on the 1st day and the deposit on the other account on the next day.

**Comment by TomAff** [comment]
@rmkhalfay: It could be, that you get problems with my method when you export the notes-field with word wraps

**Comment by rmkhalfay** [comment]
Yes,  for notes field; but one thing is creating problem for me is the balance computation for transfer fields, with withdrawal or deposit it is easy to compute, but with transfer not clear to add or subtract from previous balance in Excel? Any solution for this? Thanks in advance.

**Comment by TomAff** [comment]
Have a look on the table checkingaccount_v1
Transcode transfer
=> accoutid is the account with minus value
=> toaccountid is the account with the plus value

**Comment by vomikan** [comment]
https://github.com/moneymanagerex/moneymanagerex/issues/4097
